test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -254,5 +261,43 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
                 return base.SendAsync(request, cancellationToken);
             }
         }
+
+        private class TestStartup
+        {
+            private Startup _startup;
+            private readonly PostConfigureServices _postConfigure;
+
+            public TestStartup(IConfiguration configuration, PostConfigureServices postConfigure)
+            {
+                _startup = new Startup(configuration);
+                _postConfigure = postConfigure;
+            }
+
+            public void ConfigureServices(IServiceCollection services)
+            {
+                _startup.ConfigureServices(services);
+                _postConfigure?.ConfigureServices(services);
```

Comment:
Lets us register a callback that will be called after the call to the product's `Startup.ConfigureServices()`. This lets us get the last call to override any service in the WebHost, yet still maintain the product's setup logic.